includes/currency/class-money-formatter.php

```diff
@@ -113,6 +113,37 @@ public function format_for_display( $decimals = true ) {
 		return $this;
 	}
 
+	/**
+	 * Formats the amount for typed data returns.
+	 * Does not apply the currency code and returns a foat instead of a string.
+	 *
+	 * @since 3.0
+	 * @return Money_Formatter
+	 */
+	public function format_for_typed( $decimals = true ) {
+		$amount = $this->unformat();
+
+		if ( empty( $amount ) ) {
+			$amount = 0;
+		}
+
+		/**
+		 * Filter number of decimals to use for formatted amount
+		 *
+		 * @since unknown
+		 *
+		 * @param int        $number        Default 2. Number of decimals.
+		 * @param int|string $amount        Amount being formatted.
+		 * @param string     $currency_code Currency code being formatted.
+		 */
+		$decimals = apply_filters( 'edd_format_amount_decimals', $decimals ? $this->currency->number_decimals : 0, $amount, $this->currency->code );
```

Comment:
I meant to ask this yesterday but we got derailed by the formatting thing in Chrome.

Can we pull this `apply_filters` out into a separate private method here, since it's now applied identically in this method and `format_for_display`? I don't love how much duplication there is between the two methods, and wondered initially if the latter could use the new function and then return the customized output, but the final filters seem to make that not possible.

Reply:
Done in https://github.com/awesomemotive/easy-digital-downloads/pull/9362/commits/e3ec422b9bcf853764fd5c3e9a71220d40d2d2ef